processor/src/main/java/org/mapstruct/ap/PrimitiveToBigDecimalConversion.java

```diff
@@ -0,0 +1,61 @@
+/**
+ *  Copyright 2012-2017 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap;
+
+import static org.mapstruct.ap.internal.util.Collections.asSet;
+
+import java.math.BigDecimal;
+import java.util.Set;
+import org.mapstruct.ap.internal.conversion.SimpleConversion;
+
+import org.mapstruct.ap.internal.model.common.ConversionContext;
+import org.mapstruct.ap.internal.model.common.Type;
+
+/**
+ * Conversion between {@link BigDecimal} and native number types.
+ *
+ * @author Gunnar Morling
+ */
+public class PrimitiveToBigDecimalConversion extends SimpleConversion {
+
+    private final Class<?> targetType;
+
+    public PrimitiveToBigDecimalConversion(Class<?> targetType) {
+        if ( !targetType.isPrimitive() ) {
+            throw new IllegalArgumentException( targetType + " is no primitive type." );
+        }
+
+        this.targetType = targetType;
+    }
+
+    @Override
+    public String getToExpression(ConversionContext conversionContext) {
+        return "<SOURCE>." + targetType.getName() + "Value()";
+    }
+
+    @Override
+    public String getFromExpression(ConversionContext conversionContext) {
+        return "BigDecimal.valueOf( <SOURCE> )";
```

Comment:
I introduced with one of the latest PRs some adjustments here. You could better use `ConversionUtils.bigDecimal(conversionContext)` instead of `"BigDecimal"` to be sure that the FQN will be used if required.

Reply:
no problem. Thought I adapted them all.